aiohttp/connector.py

```diff
@@ -125,7 +125,15 @@ def __del__(self, _warnings: Any = warnings) -> None:
             if self._loop.is_closed():
                 return
 
-            self._connector._release(self._key, self._protocol, should_close=True)
+            # __del__ may be called in different thread. We have to use call_soon_threadsafe.
+            self._loop.call_soon_threadsafe(
+                functools.partial(
+                    self._connector._release,
+                    self._key,
+                    self._protocol,
+                    should_close=True,
+                )
+            )
```

Comment:
```suggestion
            self._connector._release(self._key, self._protocol, should_close=True)
```

can be fixed in a different PR now